includes/class-wcs-object-data-cache-manager.php

```diff
@@ -196,6 +215,57 @@ public function action_object_cache_changes( $object ) {
 		}
 	}
 
+	/**
+	 * When an order is restored from the trash, action on object changes.
+	 *
+	 * @param int $order_id The order id being restored.
+	 */
+	public function untrashed( $order_id ) {
+		$order = wc_get_order( $order_id );
```

Comment:
This class's name (`WCS_Object_Data_Cache_Manager`) is purposefully unspecific about the object type that it operates with. It's designed to work with all `WC_Data` types but for our purposes is only being used for orders and subscriptions.

With that in mind, I'd suggest we try to keep this class not tied to a specific object type. 

So what are your thoughts about this: 

1. Rename this and the `$order_id` references below to `$object_id` or just `$id`.
2. While `wc_get_order()` is capable of returning a subscription or order object, it is specific to the object type too.  To get around that, we could introduce a new private function called `get_object( $id )` which looks something like this:

```php
private function get_object( $id ) {
    switch ( $this->object_type ) {
        case 'order': 
           return wc_get_order( $id );
        case 'subscription': 
           return wcs_get_subscription( $id );
        default:
            return apply_filters( "wcs_object_data_cache_manager_get_{$this->object_type}_object", null, $id );
    }
}
```

Reply:
That is good to know. I see now that the term 'object' is used instead of 'order'. I have addressed this in [6872a4c](https://github.com/Automattic/woocommerce-subscriptions-core/pull/383/commits/6872a4cf76152564294d2d41ed165c81979579bf).